tokio-postgres/src/replication.rs

```diff
@@ -0,0 +1,89 @@
+use crate::client::{InnerClient, Responses};
+use crate::codec::FrontendMessage;
+use crate::connection::RequestMessages;
+use crate::{simple_query, Error};
+use bytes::{Bytes, BytesMut};
+use futures::{ready, Stream};
+use log::trace;
+use pin_project_lite::pin_project;
+use postgres_protocol::message::backend::Message;
+use postgres_protocol::message::frontend;
+use std::marker::PhantomPinned;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use std::time::{SystemTime, UNIX_EPOCH};
+const J2000_EPOCH_GAP: u128 = 946_684_800_000_000;
+pub async fn start_replication(
+    client: &InnerClient,
+    query: &str,
+) -> Result<ReplicationStream, Error> {
+    trace!("executing start replication query {}", query);
+
+    let buf = simple_query::encode(client, query)?;
+    let responses = start(client, buf).await?;
+    Ok(ReplicationStream {
+        responses,
+        _p: PhantomPinned,
+    })
+}
+
+pub async fn stop_replication(client: &InnerClient) -> Result<(), Error> {
+    trace!("executing stop replication");
+    let mut buf = BytesMut::new();
+    frontend::copy_done(&mut buf);
+    let _ = client.send(RequestMessages::Single(FrontendMessage::Raw(buf.freeze())))?;
+    Ok(())
+}
+
+pub async fn standby_status_update(
```

Comment:
Did you consider also adding support for:

* identify_system
* base_backup
* hot standby feedback
* creating/dropping slots
* timeline history
* show

(I'm not the crate maintainer so don't take these as blockers. It might be fine to do those in later PRs after the basic support is in.)